frontend/document/documentregistry.lua

```diff
@@ -4,12 +4,12 @@ This is a registry for document providers
 
 local DocSettings = require("docsettings")
 local logger = require("logger")
-local lfs = require("libs/libkoreader-lfs")
 local util = require("util")
 
 local DocumentRegistry = {
     registry = {},
     providers = {},
+    providers_hash = {}, -- hash table of registered providers { provider_key = provider }
```

Comment:
`providers_hash` sounds like it would be a hash value (a string or a number).
It is a Lua hash (not an array) table, ok, but we don't advertize that fact in most of our names :)
So, may be `known_providers` or  `provider_names` ?